pkg/processor/build/runtime/python/runtime.go

```diff
@@ -119,13 +126,51 @@ func (p *python) GetProcessorDockerfileInfo(onbuildImageRegistry string) (*runti
 		},
 	}
 
+	// copy pip ca artifact to function image
+	if runtimeConfig != nil && runtimeConfig.Python != nil {
+		if runtimeConfig.Python.PipCAPath != "" {
+			processorDockerfileInfo.Directives["preCopy"] = append(processorDockerfileInfo.Directives["preCopy"],
+				functionconfig.Directive{
+					Kind:  "COPY",
+					Value: fmt.Sprintf("%s %s", path.Base(pipCAFileLocation), pipCAFileLocation),
+				})
+		}
+	}
+
 	return &processorDockerfileInfo, nil
 }
 
-// GetBuildArgs returns building arguments
+// GetRuntimeBuildArgs returns python specific build args directives
 func (p *python) GetRuntimeBuildArgs(runtimeConfig *runtimeconfig.Config) map[string]string {
+	buildArgs := p.AbstractRuntime.GetRuntimeBuildArgs(runtimeConfig)
+
+	if runtimeConfig != nil && runtimeConfig.Python != nil {
+
+		// enrich build arg with pip ca file path
+		if runtimeConfig.Python.PipCAPath != "" {
+			buildArgs["PIP_CERT"] = pipCAFileLocation
+		}
+
+		// enrich build args with runtime specific build args
+		for key, value := range runtimeConfig.Python.BuildArgs {
+			buildArgs[key] = value
+		}
+	}
+	return buildArgs
+}
+
+func (p *python) OnAfterStagingDirCreated(runtimeConfig *runtimeconfig.Config, stagingDir string) error {
 	if runtimeConfig != nil && runtimeConfig.Python != nil {
-		return runtimeConfig.Python.BuildArgs
+		PipCAContents, err := runtimeConfig.Python.GetPipCAContents()
+		if err != nil {
+			return errors.Wrap(err, "Failed to get pip ca contents")
+		}
+
+		destPath := path.Join(stagingDir, path.Base(pipCAFileLocation))
+		p.Logger.DebugWith("Writing pip ca contents", "destPath", destPath)
+		if err := ioutil.WriteFile(destPath, PipCAContents, 0644); err != nil {
+			return errors.Wrap(err, "Failed to write pip ca contents to file")
+		}
 	}
-	return p.AbstractRuntime.GetRuntimeBuildArgs(runtimeConfig)
+	return nil
```

Comment:
Do we not need this anymore?

Reply:
actually its abstract wasnt doing anything but is it correct to call it for the day it would.